### DISCUSSION TITLE
Designer source code - Blazor, Stencil or both?

### DISCUSSION THREAD

**Discussion by briannoyes** [elsa-workflows/elsa-core#4931] [Design & tech]
I'm a little unclear about the source code and architecture for the designer if we wanted to integrate it into our own front end prior to that becoming a supported feature. I found some information that pointed [here](https://github.com/elsa-workflows/elsa-core/tree/master/src/designer) for the designer implementation, and see that is done in Stencil. But then all the current docs only mention the Blazor components. 

Is the designer being rewritten from scratch in Blazor and the Stencil one is abandoned? Or is the stencil implementation being wrapped by a Blazor component?

Appreciate any information you can provide there.

**Comment by glime-ai** [comment]
The Elsa Workflow designer is indeed being transitioned from Stencil to Blazor. This is part of an effort to provide a more integrated experience within the .NET ecosystem, particularly for Blazor applications. The current documentation primarily focuses on Blazor components because that is the direction the Elsa team is moving towards.

From the information available, it seems that the Stencil version of the designer is not being actively developed anymore, and the team is putting their energy into the new Blazor version. This new version will include a feature called CustomElements host, which will make it easier to integrate the designer into other applications.

However, until the Blazor version is fully ready and considered stable for production use, the Stencil version remains available. If you are looking to integrate the Elsa designer into your own frontend, you may need to consider whether to use the current Stencil version or wait for the Blazor version to be completed.

For the most up-to-date information, you should refer to the official Elsa documentation and GitHub repository issues/discussions, where you can find discussions about the current state and future plans for the Elsa designer.

**Comment by sfmskywalker** [comment]
The bot is correct; we have transitioned away from the Stencil implementation. The components to integrate with your own front-end are provided as Blazor components that you can use directly in case your front end is also a Blazor app, or through its custom elements wrappers, allowing you to embed them inside of any other stack used for your front-end. Documentation is still lacking, but we have a POC in the elsa-studio repo: https://github.com/elsa-workflows/elsa-studio/tree/main/src/wrappers

**Comment by briannoyes** [comment]
Thanks, appreciate the confirmation, Is there any rough roadmap for when that will be completed and part of the platform? Thanks

**Reply by sfmskywalker** [reply]
I couldn't say. Most of the roadmap is influenced by paying customers with a vested interest in certain features. Everything else comes from my spare time. That being said, I am eager to get this feature completed & documented, so it is on the top of my list of things to do whenever I have enough bandwidth.

**Comment by GeorgeS2019** [comment]
@briannoyes 

I have listed the ElsaStudio API here
https://github.com/elsa-workflows/elsa-studio/discussions/10#discussioncomment-9931646
